tools/test-proxy/Azure.Sdk.Tools.TestProxy.Tests/AdminFunctionalityTests.cs

```diff
@@ -72,35 +177,105 @@ public void TestAddTransform()
                     HttpContext = httpContext
                 }
             };
+            await controller .AddSanitizer();
 
-            testRecordingHandler.Transforms.Clear();
+            var result = testRecordingHandler.PlaybackSessions[recordingId].AdditionalSanitizers.First();
+            Assert.True(result is HeaderRegexSanitizer);
+        }
 
-            // act
-            controller.AddTransform();
+        [Fact]
+        public async void TestAddSanitizerThrowsOnBadRecordingId()
+        {
+            RecordingHandler testRecordingHandler = new RecordingHandler(Directory.GetCurrentDirectory());
+            var httpContext = new DefaultHttpContext();
+            httpContext.Request.Headers["x-recording-id"] = "bad-recording-id";
+            httpContext.Request.Headers["x-abstraction-identifier"] = "HeaderRegexSanitizer";
+            httpContext.Request.Body = TestHelpers.GenerateStreamRequestBody("{ \"key\": \"Location\", \"value\": \"https://fakeazsdktestaccount.table.core.windows.net/Tables\" }");
+            httpContext.Request.ContentLength = 92;
 
-            var result = testRecordingHandler.Transforms.First();
+            var controller = new Admin(testRecordingHandler)
+            {
+                ControllerContext = new ControllerContext()
+                {
+                    HttpContext = httpContext
+                }
+            };
 
-            // assert
-            Assert.True(result is ApiVersionTransform);
+            await Assert.ThrowsAsync<BadHttpRequestException>(
+               async () => await controller.AddSanitizer()
+            );
         }
 
         [Fact]
-        public void TestAddTransformIndividualRecording()
+        public async void TestAddTransform()
```

Comment:
```suggestion
        public async Task TestAddTransform()
```

Reply:
I'd suggest changing these and all other async tests to return Task in case returning void causes any non-determinism with the xUnit test runner.